core/src/main/scala/org/apache/spark/scheduler/cluster/mesos/MesosSchedulerBackend.scala

```diff
@@ -124,6 +124,13 @@ private[spark] class MesosSchedulerBackend(
         Value.Scalar.newBuilder()
           .setValue(MemoryUtils.calculateTotalMemory(sc)).build())
       .build()
+
+    sc.conf.getOption("spark.mesos.uris").map { uris =>
+      uris.split(",").foreach { uri =>
+        command.addUris(CommandInfo.URI.newBuilder().setValue(uri.trim()))
+      }
+    }
```

Comment:
this is duplicated 3 times. Do you think the refactoring your other patch (roles principals) will help? Should we merge that one first?